test/1_stdlib/Inputs/ProcessStressTest/ProcessStressTest.c

```diff
@@ -0,0 +1,48 @@
+#include <CoreFoundation/CoreFoundation.h>
+#include <dispatch/dispatch.h>
+#include "ProcessStressTest.h"
+#include <pthread.h>
+#include <assert.h>
+
+static pthread_mutex_t mutex = PTHREAD_MUTEX_INITIALIZER;
+static pthread_mutex_t waiterMutex = PTHREAD_MUTEX_INITIALIZER;
+
+static void run_process_stress_test(void) {
+  dispatch_queue_t queue = dispatch_queue_create(nil, DISPATCH_QUEUE_CONCURRENT);
+  dispatch_group_t group = dispatch_group_create();
+  pthread_mutex_init(&mutex, NULL);
+  pthread_mutex_init(&waiterMutex, NULL);
+
+  CFMutableSetRef results = CFSetCreateMutable(NULL, 0, &kCFTypeSetCallBacks);
+  pthread_mutex_lock(&waiterMutex);
```

Comment:
Instead of the waiter mutex, you can actually pause the queue. See [`dispatch_suspend`](https://developer.apple.com/reference/dispatch/1452801-dispatch_suspend).